django/contrib/admin/widgets.py

```diff
@@ -382,3 +384,121 @@ def __init__(self, attrs=None):
 
 class AdminBigIntegerFieldWidget(AdminIntegerFieldWidget):
     class_name = 'vBigIntegerField'
+
+
+class AutocompleteMixin:
+    """
+    ForeignKey widget that loads options via AJAX to avoid rendering large Querysets.
+
+    This class is responsible for rendering the necessary
+    data attributes for select2 as well as adding the static
+    form media.
+    """
+
+    url_pattern = 'admin:%s_%s_autocomplete'
+
+    def __init__(self, model, attrs=None, choices=(), using=None):
+        self.model = model
+        self.db = using
+        self.choices = choices
+        if attrs is not None:
+            self.attrs = attrs.copy()
+        else:
+            self.attrs = {}
+
+    def get_url_name(self):
+        return self.url_pattern % (self.model_admin.model._meta.app_label, self.model.__name__)
```

Comment:
Cannot this information solely be extracted from the model itself? ie `.model._meta.app_label`.